src/develop/imageop_gui.h

```diff
@@ -26,6 +26,20 @@ GtkWidget *dt_bauhaus_combobox_from_params(dt_iop_module_t *self, const char *pa
 
 GtkWidget *dt_bauhaus_toggle_from_params(dt_iop_module_t *self, const char *param);
 
+typedef struct dt_iop_module_section_t
+{
+  dt_action_type_t actions;
+  dt_iop_module_t *self;
+  gchar *section;
+} dt_iop_module_section_t;
+
+/*
+ * package dt_iop_module_t pointer and section name to pass to a _from_params function
+ * it will then create a widget action in a section, rather than top level in the module
+ */
+#define DT_IOP_SECTION_FOR_PARAMS(self, section) \
+    (dt_iop_module_t *)&(dt_iop_module_section_t){DT_ACTION_TYPE_IOP_SECTION, self, section}
```

Comment:
I think this deserves some comment that this struct (dt_iop_module_section_t) must start with actions as it is overlaid over a dt_iop_module_t and that actions is checked to know if it is a real module or a section (DT_ACTION_TYPE_IOP_SECTION).

Reply:
I've now added the same comment that the other similar action types have 
`// !!! NEEDS to be FIRST (to be able to cast convert)`

BTW notice that this case is very similar to `dt_iop_module_t` itself, which also only contains a marker `DT_ACTION_TYPE_IOP_INSTANCE` at the start, so that it can be used to initialise actions linked to a module (`DT_ACTION_TYPE_IOP`) but where the widget belongs to an instance. This is the reason that we can't just use `dt_action_section` here to have _from_params_ put the widgets in a section of the module, because we still need to know which module _instance_ they belong to.